spec/lib/jobs/dell-wsman-simple-update-firmware-spec.js

```diff
@@ -0,0 +1,278 @@
+            fs.readFile.yields(null, fakeXml);
+            return simpleUpdateFirmwareJob.getComponentType().then(function() {
+                expect(simpleUpdateFirmwareJob.componentIds).to.include('159');
+                expect(ChildProcess.prototype.run).to.have.been.called.twice;
+            });
+        });
+
+        it('should create tmp folder and get component id correctly', function() {
+            fs.existsSync.withArgs('/tmp').returns(false);
+            fs.existsSync.withArgs('/tmp/fdc41b55-d61d-45c1-9b67-651922e94daa').returns(false);
+            fs.existsSync.withArgs('/tmp/SAS-RAID_Firmware_03NXN_WN64_25.5.2.0001_A07_01.EXE').returns(false);
+            ChildProcess.prototype.run.resolves();
+            var fakeXml = '<Device componentID="159" embedded="1"><Display lang="en"><![CDATA[PowerEdge C6320 BIOS]]></Display></Device>';
+            fs.readFile.yields(null, fakeXml);
+            return simpleUpdateFirmwareJob.getComponentType().then(function() {
+                expect(simpleUpdateFirmwareJob.componentIds).to.include('159');
+                expect(ChildProcess.prototype.run).to.have.been.called.quartic;
+            });
+        });
+
+        it('should handle: error occurs while reading package.xml', function() {
+            fs.existsSync.withArgs('/tmp').returns(false);
+            fs.existsSync.withArgs('/tmp/fdc41b55-d61d-45c1-9b67-651922e94daa').returns(false);
+            fs.existsSync.withArgs('/tmp/SAS-RAID_Firmware_03NXN_WN64_25.5.2.0001_A07_01.EXE').returns(false);
+            ChildProcess.prototype.run.resolves();
+            fs.readFile.yields(new Error('fake'), '');
+            expect(simpleUpdateFirmwareJob.getComponentType()).to.be.rejectedWith('An error occurred while reading package xml file.');
+        });
+
+        it('should create tmp folder and get component id correctly', function() {
+            fs.existsSync.withArgs('/tmp').returns(false);
+            fs.existsSync.withArgs('/tmp/fdc41b55-d61d-45c1-9b67-651922e94daa').returns(false);
+            fs.existsSync.withArgs('/tmp/SAS-RAID_Firmware_03NXN_WN64_25.5.2.0001_A07_01.EXE').returns(false);
+            ChildProcess.prototype.run.resolves();
+            fs.readFile.yields(null, '<SoftwareComponent dateTime="2017-01-10T02:46:46-06:00"></SoftwareComponent>');
+            expect(simpleUpdateFirmwareJob.getComponentType()).to.be.rejectedWith('Could not found any device tag in package.xml.');
+        });
+    });
+
+    describe('_handleAsyncRequest', function(){
+        it('should send request successfully', function() {
+            simpleUpdateFirmwareJob.firmwareConfigs = {
+                dupUpdater: 'http://localhost:46010/api/1.0/server/firmware/updater/dup'
+            };
+            simpleUpdateFirmwareJob.componentIds = [ '159' ];
+            this.sandbox.stub(SimpleUpdateFirmwareJob.prototype, 'clientRequest').resolves();
+            expect(simpleUpdateFirmwareJob._handleAsyncRequest()).to.be.fulfilled;
```

Comment:
The promise need be returned, otherwise this test case will always succeed no matter the promise is rejected or fulfilled.